clouddq/runners/dbt/dbt_runner.py

```diff
@@ -0,0 +1,233 @@
+                gcp_region_id=gcp_region_id,
+                gcp_bq_dataset_id=gcp_bq_dataset_id,
+                gcp_service_account_key_path=gcp_service_account_key_path,
+                gcp_impersonation_credentials=gcp_impersonation_credentials,
+            )
+            self.connection_config = connection_config
+            self.dbt_profiles_dir = Path(self.dbt_path)
+            logger.debug(
+                "Writing user input GCP connection profile to dbt profiles.yml "
+                f"at path: {self.dbt_profiles_dir}",
+            )
+            if environment_target:
+                logger.debug(f"Using `environment_target`: {environment_target}")
+                self.environment_target = environment_target
+                self.connection_config.to_dbt_profiles_yml(
+                    target_directory=self.dbt_profiles_dir,
+                    environment_target=self.environment_target,
+                )
+            else:
+                self.environment_target = DEFAULT_DBT_ENVIRONMENT_TARGET
+                self.connection_config.to_dbt_profiles_yml(
+                    target_directory=self.dbt_profiles_dir
+                )
+
+    def __resolve_dbt_path(
+        self,
+        dbt_path: str,
+        create_paths_if_not_exists: bool = False,
+        write_log: bool = False,
+    ) -> Path:
+        logger.debug(f"Current working directory: {Path().cwd()}")
+        if not dbt_path:
+            dbt_path = Path().cwd().joinpath("dbt").absolute()
+            logger.debug(
+                "No argument 'dbt_path' provided. Defaulting to use "
+                f"'dbt' directory in current working directory at: {dbt_path}"
+            )
+        else:
+            dbt_path = Path(dbt_path).absolute()
+            if dbt_path.name != "dbt":
+                dbt_path = dbt_path / "dbt"
+        if not dbt_path.is_dir():
+            if create_paths_if_not_exists:
+                logger.debug(f"Creating a new dbt directory at 'dbt_path': {dbt_path}")
+                dbt_path.mkdir(parents=True, exist_ok=True)
+            else:
+                raise ValueError(f"Provided 'dbt_path' does not exists: {dbt_path}")
+        if write_log:
+            logger.debug(f"Using 'dbt_path': {dbt_path}")
+        return dbt_path
+
+    def __prepare_dbt_project_path(self) -> None:
+        dbt_project_path = self.dbt_path.absolute().joinpath("dbt_project.yml")
+        if not dbt_project_path.is_file():
+            logger.debug(
+                f"Cannot find `dbt_project.yml` in path: {dbt_project_path} \n"
+                f"Writing templated file to: {dbt_project_path}/dbt_project.yml"
+            )
+            write_templated_file_to_path(dbt_project_path, DBT_TEMPLATED_FILE_LOCATIONS)
+        logger.debug(f"Using 'dbt_project_path': {dbt_project_path}")
+
+    def __prepare_dbt_main_path(self) -> None:
+        assert self.dbt_path.is_dir()
+        dbt_main_path = self.dbt_path / "models" / "data_quality_engine"
```

Comment:
Can we create constants for static strings?

Reply:
alas Python doesn't have constants :( declaring these variables at global scopes means they could be changed at runtime which we do not want

Reply:
can we create a separate constants.py file and import these required variables from the same? we can discuss.